src/plugin.ts

```diff
@@ -32,6 +32,8 @@ export class PluginTypeEnvironment extends TypeEnvironment {
 
     public drawableTypes: Map<CoreElementKind, ObjectType>;
 
+    public startTypes: [Type[], Type][];
```

Comment:
This is a pretty confusing data structure to those not immersed in TSC. Might I suggest wrapping it in a class? Something like this

```typescript
interface ProgramSignature {
  overloadCount: number;
  overloads: [Type[], Type][];
}
```
Mostly the same, but more explicit naming would make it easier to understand.